hcl2template/utils.go

```diff
@@ -187,3 +188,50 @@ func ConvertPluginConfigValueToHCLValue(v interface{}) (cty.Value, error) {
 	}
 	return buildValue, nil
 }
+
+// GetVarsByType walks through a hcl body, and gathers all the Traversals that
+// have a root type matching one of the specified top-level labels.
+//
+// This will only work on finite, expanded, HCL bodies.
+//
+// TODO: make it work on JSON serialised HCL2 templates.
```

Comment:
Is this still true? If so could you just add a little context to the comment on what doesn't work

Reply:
Oh no this is very much out-of-date, JSON templates work fine since there's no concept of block in JSON, so the whole tree gets visited at once when getting the variables from an expression.